### PR TITLE
OnlineDQM : Tune L1T Muon QTs v2

### PR DESCRIPTION
**Pull request by stahlleiton**
Description:

This PR includes the following changes:

- Reduce the warning threshold of the L1T muon BX QTs from 60% to 50% for all TF, and also slightly reduce the threshold of the noisy channel QT for EMTF.

- Modify the modules that make the muon comparison plots ( L1TStage2RegionalMuonCandComp and  L1TStage2RegionalMuonCandComp )  to include the option of ignoring bins when making the muon mismatch plots, so that they don't take into account the ignored categories when summing up the total number of mismatch muons. The list of ignored bins is then properly propagated to the modules that compute the mismatch ratio plots.

- Include the PeakAtBX0 Quality Tests for muons in the L1T Summary plot

### CODE REVIEW

**Comment by dmitrijus** [comment]
+1

**Comment by stahlleiton** [comment]
Was trying to check the log file of the step 4 of the relval 1306.0 , but I don't see or understand what could have gone wrong.

**Comment by dmitrijus** [comment]
please test

File: DQM/L1TMonitor/src/L1TStage2MuonComp.cc [hunk 1]
```diff
@@ -8,8 +8,19 @@ L1TStage2MuonComp::L1TStage2MuonComp(const edm::ParameterSet& ps)
       muonColl1Title(ps.getUntrackedParameter<std::string>("muonCollection1Title")),
       muonColl2Title(ps.getUntrackedParameter<std::string>("muonCollection2Title")),
       summaryTitle(ps.getUntrackedParameter<std::string>("summaryTitle")),
+      ignoreBin(ps.getUntrackedParameter<std::vector<int>>("ignoreBin")),
       verbose(ps.getUntrackedParameter<bool>("verbose"))
 {
+  // First include all bins
+  for (uint i = 1; i <= RIDX; i++) {
```
Comment (davidlange6) [inline DQM/L1TMonitor/src/L1TStage2MuonComp.cc]: hi @stahlleiton - please replace by
incBin={true};

File: DQM/L1TMonitor/src/L1TStage2MuonComp.cc [hunk 2]
```diff
@@ -68,6 +80,13 @@ void L1TStage2MuonComp::bookHistograms(DQMStore::IBooker& ibooker, const edm::Ru
   errorSummaryNum->setBinLabel(RISO, "iso mismatch", 1);
   errorSummaryNum->setBinLabel(RIDX, "index mismatch", 1);
 
+  // Change the label for those bins that will be ignored
+  for (uint i = 1; i <= RIDX; i++) {
```
Comment (davidlange6) [inline DQM/L1TMonitor/src/L1TStage2MuonComp.cc]: avoid uint - use unsigned int

File: DQM/L1TMonitor/src/L1TStage2MuonComp.cc [hunk 3]
```diff
@@ -216,60 +235,74 @@ void L1TStage2MuonComp::analyze(const edm::Event& e, const edm::EventSetup& c) {
...
 
-      bool muonMismatch = false;
+      bool muonMismatch = false;    // All muon mismatches
+      bool muonSelMismatch = false; // Muon mismatches excluding ignored bins
       if (muonIt1->hwPt() != muonIt2->hwPt()) {
         muonMismatch = true;
+        if (incBin[RPT]) muonSelMismatch = true;
         summary->Fill(PTBAD);
-        errorSummaryNum->Fill(RPT);
+        if (incBin[RPT]) errorSummaryNum->Fill(RPT);
```
Comment (davidlange6) [inline DQM/L1TMonitor/src/L1TStage2MuonComp.cc]: combine with the if on line 242 (and similarly below, avoid the duplicated ifs)

File: DQM/L1TMonitor/src/L1TStage2MuonComp.cc (hunk 1 above)
Comment (stahlleiton) [inline DQM/L1TMonitor/src/L1TStage2MuonComp.cc]: Will this work for an array?
I tried to change it, but got the following error:
error: assigning to an array from an initializer list

Comment (davidlange6) [inline DQM/L1TMonitor/src/L1TStage2MuonComp.cc]: ah - incBin is a member variable..ok - never mind this (but best to make your local variables via some naming convention - typically a trailing _.

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**